core/src/main/java/org/infinispan/commands/functional/ReadOnlyManyCommand.java

```diff
@@ -9,38 +9,67 @@
 import java.util.Collection;
 import java.util.function.Function;
 
+import org.infinispan.cache.impl.CacheEncoders;
+import org.infinispan.cache.impl.EncodingClasses;
 import org.infinispan.commands.AbstractTopologyAffectedCommand;
 import org.infinispan.commands.LocalCommand;
 import org.infinispan.commands.Visitor;
-import org.infinispan.functional.EntryView.ReadEntryView;
 import org.infinispan.commons.marshall.MarshallUtil;
 import org.infinispan.container.entries.CacheEntry;
 import org.infinispan.context.InvocationContext;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.functional.EntryView.ReadEntryView;
 import org.infinispan.functional.impl.EntryViews;
 import org.infinispan.functional.impl.Params;
+import org.infinispan.marshall.core.EncoderRegistry;
 
 public class ReadOnlyManyCommand<K, V, R> extends AbstractTopologyAffectedCommand implements LocalCommand {
    public static final int COMMAND_ID = 63;
 
    protected Collection<? extends K> keys;
    protected Function<ReadEntryView<K, V>, R> f;
    protected Params params;
-
-   public ReadOnlyManyCommand(Collection<? extends K> keys, Function<ReadEntryView<K, V>, R> f, Params params) {
+   private EncodingClasses encodingClasses;
+   protected transient CacheEncoders cacheEncoders = new CacheEncoders();
+
+   public ReadOnlyManyCommand(Collection<? extends K> keys,
+                              Function<ReadEntryView<K, V>, R> f,
+                              Params params,
+                              EncodingClasses encodingClasses,
+                              ComponentRegistry componentRegistry) {
       this.keys = keys;
       this.f = f;
       this.params = params;
+      this.encodingClasses = encodingClasses;
       this.setFlagsBitSet(params.toFlagsBitSet());
+      init(componentRegistry);
    }
 
    public ReadOnlyManyCommand() {
    }
 
-   public ReadOnlyManyCommand(ReadOnlyManyCommand c) {
+   public ReadOnlyManyCommand(ReadOnlyManyCommand c, ComponentRegistry componentRegistry) {
```

Comment:
This is a copy-constructor, so it shouldn't need to re-wire anything. Is there a problem with `this.cacheEncoders = c.cacheEncoders`?